### PR TITLE
Include dates in release note headings

### PR DESCRIPTION
**Pull request by lagru**
## Description

Format release note headings more consistently. I also thought including the release dates might be useful sometimes when digging through commit dates etc.

## Checklist

- A descriptive but concise pull request title
- [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- [Unit tests](https://scikit-image.org/docs/dev/development/contribute.html#testing)
- A gallery example in `./doc/examples` for new features
- [Contribution guide](https://scikit-image.org/docs/dev/development/contribute.html) is followed

## Release note

Summarize the introduced changes in the code block below in one or a few sentences. The
summary will be included in the next release notes automatically:

```release-note
...
```

### CODE REVIEW

**Review by soupault — APPROVED** [review]
LGTM, hopefully also works for the US-based devs 😅

File: doc/source/release_notes/release_template.rst [hunk 1]
```diff
@@ -1,7 +1,7 @@
 :orphan:
 
-scikit-image 0.X.0 release notes
-================================
+scikit-image 0.X.0
```
Comment (stefanv) [inline doc/source/release_notes/release_template.rst:3]: Shouldn't this get a `(date)` as well?

**Review by stefanv — APPROVED** [review]

File: doc/source/release_notes/release_template.rst (hunk 1 above)
Comment (lagru) [reply doc/source/release_notes/release_template.rst:3]: Good point.

Comment (lagru) [reply doc/source/release_notes/release_template.rst]: Though, I don't think we have much use for the template now that we are using changelist. Should we just remove it (in a follow-up PR)?

**Pull request merged**